core/domain/rights_manager_test.py

```diff
@@ -584,114 +470,62 @@ def test_ownership_of_collection(self):
-        self.assertTrue(
-            rights_manager.Actor(self.user_id_moderator).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_moderator).can_edit(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_moderator).can_delete(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_play(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_edit(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_delete(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertTrue(rights_manager.check_can_access_activity(
+            self.user_id_moderator, self.actions_moderator, collection_rights))
+        self.assertFalse(rights_manager.check_can_edit_activity(
+            self.user_id_moderator, self.actions_moderator, collection_rights))
+
+        self.assertFalse(rights_manager.check_can_access_activity(
+            self.user_id_b, self.actions_b, collection_rights))
+        self.assertFalse(rights_manager.check_can_edit_activity(
+            self.user_id_b, self.actions_b, collection_rights))
 
     def test_inviting_collaborator_to_collection(self):
         self.save_new_valid_collection(
             self.COLLECTION_ID, self.user_id_a,
             exploration_id=self.EXP_ID_FOR_COLLECTION)
+        collection_rights = rights_manager.get_collection_rights(
+            self.COLLECTION_ID)
 
         # Verify initial editor permissions for the collection.
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_play(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_view(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_edit(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
-        self.assertFalse(
-            rights_manager.Actor(self.user_id_b).can_delete(
-                constants.ACTIVITY_TYPE_COLLECTION, self.COLLECTION_ID))
+        self.assertTrue(rights_manager.check_can_access_activity(
+            self.user_id_a, self.actions_a, collection_rights))
+        self.assertTrue(rights_manager.check_can_edit_activity(
```

Comment:
I think I did that as I couldn't find the code to delete the collection anywhere. Added check_can_delete_activity here and below.

(where is the code to delete collection)